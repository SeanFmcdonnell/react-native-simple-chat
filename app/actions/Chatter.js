import MyWebSocket from '../lib/MyWebSocket';

class Chatter {

    constructor(receiveMessage) {
        this.myWebSocket = new MyWebSocket('wss://remy-ws.glitch.me/', receiveMessage);
        this.myWebSocket.init();
    }

    sendMessage(message, cb) {
        this.myWebSocket.sendMessage(message);
        cb();
    }

    static createMessage(sender, message) {
        return { who: sender, what: message };
    }

    static isMe(message) {
        return message.who === 'me';
    }

    static getMessage(message) {
        return message.what;
    }
}

export {
    Chatter
};
