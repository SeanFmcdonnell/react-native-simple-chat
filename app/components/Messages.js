import React from 'react';
import { Dimensions, ScrollView, StyleSheet, View } from 'react-native';
import { Chatter } from "../actions/Chatter";
import { Message } from "./Message";

class Messages extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <ScrollView style={ [styles.content, this.props.style] }
                        ref={ ref => this.scrollView = ref }
                        eyboardDismissMode='interactive'
                        onContentSizeChange={ (width, height) => {
                            this.scrollView.scrollToEnd();
                        } }>

                {
                    this.props.messages.length > 0 ? this.props.messages.map((message, index) => (
                            <View key={ index }
                                  style={ Chatter.isMe(message) ? styles.messageRight : styles.messageLeft }>
                                <Message text={ Chatter.getMessage(message) }
                                         align={ Chatter.isMe(message) ? 'right' : 'left' }/>
                            </View>
                        )
                    ) : null
                }
            </ScrollView>
        );
    }
}

const height = Dimensions.get('window').height;
const styles = StyleSheet.create({
    content: {
        position: 'absolute',
        top: 90,
        left: 0,
        right: 0,
        padding: 20,
        height: height - 90 - 70,
        backgroundColor: "#ffffff"
    },
    messageLeft: {
        alignItems: 'flex-start'
    },
    messageRight: {
        alignItems: 'flex-end'
    }
});

export {
    Messages
};
