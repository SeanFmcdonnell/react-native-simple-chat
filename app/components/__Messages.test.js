import React from 'react';

import renderer from 'react-test-renderer';
import { Messages } from "./Messages";
import { Chatter } from "../actions/Chatter";

test('renders correctly with no message', () => {
    const tree = renderer.create(<Messages style={ { marginTop: '0' } }
                                           messages={ [] }/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly with 1 message', () => {
    const tree = renderer.create(<Messages style={ { marginTop: '0' } }
                                           messages={ [Chatter.createMessage('me', 'hello')] }/>).toJSON();
    expect(tree).toMatchSnapshot();
});
