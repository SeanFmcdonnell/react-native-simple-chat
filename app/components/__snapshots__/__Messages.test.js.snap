// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 1 message 1`] = `
<RCTScrollView
  eyboardDismissMode="interactive"
  onContentSizeChange={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 1174,
        "left": 0,
        "padding": 20,
        "position": "absolute",
        "right": 0,
        "top": 90,
      },
      Object {
        "marginTop": "0",
      },
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "borderRadius": 12,
              "borderWidth": 0,
              "marginBottom": 5,
              "maxWidth": 200,
              "overflow": "hidden",
              "padding": 8,
              "textAlign": "left",
            },
            Object {
              "backgroundColor": "rgb(0, 122, 255)",
              "color": "white",
            },
          ]
        }
      >
        hello
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders correctly with no message 1`] = `
<RCTScrollView
  eyboardDismissMode="interactive"
  onContentSizeChange={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 1174,
        "left": 0,
        "padding": 20,
        "position": "absolute",
        "right": 0,
        "top": 90,
      },
      Object {
        "marginTop": "0",
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;
